cpp/core/memory/MemoryAllocator.cc

```diff
@@ -22,10 +22,10 @@
 namespace gluten {
 
 bool ListenableMemoryAllocator::allocate(int64_t size, void** out) {
-  listener_->allocationChanged(size);
+  listener_->reservationChanged(size);
   bool succeed = delegated_->allocate(size, out);
   if (!succeed) {
-    listener_->allocationChanged(-size);
+    listener_->reservationChanged(-size);
   }
```

Comment:
Probably the terms should be clarified:

`allocation`: the actual memory used by the listener's caller
`reservation`: all reserved memory, including the over-borrowed memory from Spark's memory manager 

So here we'd better use `allocationChanged` rather than `reservationChanged`?

Reply:
In my view this API should align with its name, ReservationListener's caller use this API to reserve some virtual bytes,  if success, then caller can allocate real bytes.

Native memory manager report it's used bytes to Spark, which means reserve bytes from Spark, real allocate bytes should happens in memory manager internal.

Reply:
> real allocate bytes should happens in memory manager internal.

That only applies for Velox backend. The base code is in `cpp/core`. Even for Velox backend it will probably not stand for long time since we should finally hook the allocations in Velox's pools rather than reservations.

Looking back to this code, the code here is for real "allocations", and the function has signature `::allocate(int64_t size, void** out)`.

Or if the codes are confusing, I'd suggest renaming `SparkReservationListener` to `SparkMemoryReporter` or something sounds more explanatory, it doesn't have to end with `Listener` based on my understanding.

Reply:
already revert.

AllocationListener also make sense for me, I have no idea for more meaningful name. :)